turbinia/workers/__init__.py

```diff
@@ -706,16 +706,13 @@ def run_wrapper(self, evidence):
         trace = traceback.format_exc()
         log.error(message)
         log.error(trace)
-        if self.result:
-          self.result.log(message, level=logging.ERROR)
-          self.result.log(trace)
-          if hasattr(exception, 'message'):
-            self.result.set_error(exception.message, traceback.format_exc())
-          else:
-            self.result.set_error(exception.__class__, traceback.format_exc())
-          self.result.status = message
+        self.result.log(message, level=logging.ERROR)
```

Comment:
Why remove the check for whether self.result is available here?  If `self.run()` returns something other than a result object running `self.result.log()` will cause other errors.

Reply:
I believe there is always a valid result if it gets here since we already check to see if there is a valid result after ```setup()``` call.

Reply:
But the call to `self.run()` on line 707 here could potentially return something other than a `TurbiniaTaskResult` object if there is a bug in the Task code (which we've actually had before when some Tasks returned `None` objects when they early exited due to failures in the Task itself).  I'm not sure if it's defined in python what will happen when there is an exception in `self.run()`.  If we can verify the behavior in this case, then it's probably fine, but if not it might be better to be safe than sorry.